File: src/main/java/me/spartacus04/jext/command/ParameterDisc.java
package me.spartacus04.jext.command;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import me.spartacus04.jext.config.ConfigDiscManager;
import org.bukkit.command.CommandSender;

class ParameterDisc extends Parameter {

  ParameterDisc(boolean required) {
    super(required);
  }

  @Override
  String getName() {
    return "namespace";
  }

  @Override
  List<String> onComplete(String parameter, CommandSender sender) {
    final Set<String> namespaces = ConfigDiscManager.getInstance().getNamespaces();
    final List<String> matches = new ArrayList<>();
    for (final String namespace : namespaces) {
      if (matches.size() >= 8) break;
      if (namespace.toLowerCase().startsWith(parameter.toLowerCase())) {
        matches.add(namespace);
      }
    }
    if (matches.size() > 0) return matches;
    return null;
  }
  
}

File: src/main/java/me/spartacus04/jext/command/ParameterDisc.kt
package me.spartacus04.jext.command

import me.spartacus04.jext.config.ConfigData.Companion.DISCS
import me.spartacus04.jext.config.Disc
import org.bukkit.command.CommandSender

internal class ParameterDisc(required: Boolean) : Parameter(required) {
    override val name: String
        get() = "namespace"

    override fun onComplete(parameter: String, sender: CommandSender): List<String>? {
        val discs = DISCS.map { it.DISC_NAMESPACE }

        val matches = discs.filter { it.startsWith(parameter, true) }

        return matches.ifEmpty { return null }
    }

    companion object {
        fun getDisc(parameter: String) : Disc? {
            return DISCS.find { it.DISC_NAMESPACE == parameter }
        }
    }
}